promotion-guide.adoc

```diff
@@ -0,0 +1,137 @@
+
+## Release bundles
+
+A specific "release bundle" is identified by four data points: a `version` (as outlined above), a `changeId`, a build number and an environment. The `version`, `changeId` and `environment` are user-supplied input parameters to the MRO pipeline, the build number is calculated automatically. The `changeId` can be any string meaningful to the user, its value does not have any effect on the operation of the MRO pipeline. The environment input variable (such as `DEV`) will be shortened to a single-letter token (e.g. `D`).
+
+Technically speaking, a release bundle is a certain state of the release manager repository and the state of each linked repository at that time. This state is identified by a Git tag. For example, a release bundle with `version=1`, `changeId=1234`, `buildNumber=0` and `environment=DEV` is identified by the Git tag `v1-1234-0-D`. This tag is set on the release manager repository, and all repositories the `metadata.yml` refers to at this time.
+
+## Environments
+
+The MRO assumes three "conceptual" environments: DEV, QA and PROD (with short token forms D, Q and P). Those environments are strictly ordered - a state should go from DEV to QA, and then from QA to PROD.
+
+To ensure that software progresses along the DEV -> QA -> PROD path, release bundles from environment DEV can only be installed into QA, and only a release bundle from QA can be installed into PROD. Installing a release bundle from DEV into PROD is not allowed.
+
+Each "conceptual" environment is mapped to an OpenShift namespace:
+
+- DEV to `$PROJECT-dev` (e.g. `foo-dev`)
+- QA to `$PROJECT-test` (e.g. `foo-test`. Note that it is NOT `-qa`!)
+- PROD to `$PROJECT-prod` (e.g. `foo-prod`)
+
+Keep in mind that when you create a new project with OpenDevStack, you get three OpenShift namespaces:
+
+- `foo-dev` (your DEV environment)
+- `foo-test` (your QA environment - unfortunately not named `-qa` for historical reasons)
+- `foo-cd` (where Jenkins runs and the pipelines such as the MRO pipeline are executed)
+
+So while there is a corresponding namespace for DEV and QA, there is no namespace corresponding to the PROD environment out-of-the-box. This is because it is assumed that your PROD environment is likely on another cluster altogether. To create `foo-prod` on another cluster, you (or someone with appropriate rights) can run the script located at https://github.com/opendevstack/ods-core/blob/master/ocp-scripts/create-target-project.sh. Then you need to tell MRO two things: where the API of the external cluster is, and the credentials with which to access it. A typical configuration is:
```

Comment:
A side note to whether this applies to on prem clusters would be appreciated.

Reply:
The comment applies to "ODS clusters" - not taking into account where they are deployed. Given that this is an open source project, I'll keep it as-is.